src/Plutip/Server.purs

```diff
@@ -143,12 +154,22 @@ withPlutipContractEnv plutipCfg distr cont = do
     withResource (startCtlServer plutipCfg)
       (stopChildProcessWithPort plutipCfg.ctlServerConfig.port) <<< const
 
-  withWallets :: ClusterStartupParameters -> (wallets -> Aff a) -> Aff a
-  withWallets response cc = case decodeWallets response.privateKeys of
-    Nothing ->
-      liftEffect $ throw $ "Impossible happened: unable to decode" <>
-        " wallets from private keys. Please report as bug."
-    Just wallets -> cc wallets
+  withWallets
+    :: ContractEnv ()
+    -> PrivatePaymentKey
+    -> ClusterStartupParameters
+    -> (wallets -> Aff a)
+    -> Aff a
+  withWallets env ourKey response cc = do
+    wallets <- runContractInEnv env do
```

Comment:
I wonder if we should change the log level before we run this, so the user doesn't get flooded with our traces

Reply:
We should provide a `customLogger` that logs nothing.